src/race_logic.c

```diff
@@ -936,16 +906,15 @@ void func_8028F970(void) {
             var_s2 += 1;
             var_s3 += 0xDD8;
             if (var_s2 == 4) {
-                if ((gEnableDebugMode != 0) && (gModeSelection != BATTLE)) {
-                    temp_a1 = gControllerOne;
-                    if (temp_a1->unk6 & 0x800) {
+                if ((gEnableDebugMode != 0) && (gModeSelection != 3)) {
+                    if (gControllerOne->buttonPressed & 0x800) {
                         gLapCountByPlayerId->unk0 = 2;
                     }
-                    if (temp_a1->unk6 & 0x100) {
+                    if (gControllerOne->buttonPressed & 0x100) {
```

Comment:
Do you mind also replacing the 0x100 and what nots with A_BUTTON or whatever else.

Reply:
Sure

Reply:
Updated